### PR TITLE
phase2

### PR DESCRIPTION
**Pull request by aradmaleki02**

### CODE REVIEW

File: src/DataInitializer.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import java.io.File;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class DataInitializer {
+    private static final DataInitializer INSTANCE;
+
+    static {
+        INSTANCE = new DataInitializer();
+    }
```
Comment (Mir-Abedi) [inline src/DataInitializer.java:10]: static block not needed.
```suggesstion
    private static final DataInitializer INSTANCE = new DataInitializer();
```

File: src/Searcher.java [hunk 2]
```diff
@@ -26,20 +40,15 @@ public void run() {
             input = getInput();
             input = processInput(input);
             String[] words = input.split("[ ]+");
```
Comment (Mir-Abedi) [inline src/Searcher.java]: regex could be written as \\s+

File: src/Searcher.java [hunk 3]
```diff
@@ -48,19 +57,26 @@ private String processInput(String input) {
...
+        if (word.charAt(0) == '+')
+            arrayList.addAll(dataContainer.getAddress(word.deleteCharAt(0).toString()));
+        else if (word.charAt(0) == '-')
+            arrayList.removeAll(dataContainer.getAddress(word.deleteCharAt(0).toString()));
+        else
+            arrayList.retainAll(dataContainer.getAddress(word.toString()));
     }
 
     private void sortWordsQuery(String[] words) {
+        // Making "words" an arraylist to sort it
```
Comment (Mir-Abedi) [inline src/Searcher.java]: comment not needed in the code.

**Review by Mir-Abedi — APPROVED** [review]
well done! 🥇

**Comment by aradmaleki02** [comment]
> well done! 🥇

agha kamet garm

**Comment by aradmaleki02** [comment]
folder name (db) doesn't have a meaning.
SearchController line 59: probably it's better to write if and else
overall point: mentor said if your hashset is doing different stuff at parts of your code, it is better to make a class for it. it also looks better to pass an object of your own than passing an arraylist of hashset of strings.
agha aali javabetoono dadam

File: src/DataInitializer.java (hunk 1 above)
Comment (Arash1381-y) [reply src/DataInitializer.java:10]: according to my searches using the static block is not mentioned as a bad practice how ever for making instance is better to use in line initialization . 
Thank You Mr.Abedi

File: src/Searcher.java (hunk 3 above)
Comment (Arash1381-y) [reply src/Searcher.java]: thanks for mention this

**Comment by Siasor88** [comment]
**Lack of dependency managing tool!!**

**Searcher**

> -Class could be singleTone or have static members!

> -wild card import (line 1)

> -scanner could be one of the run() parameters instead of being a class field. (line 4)

> infinite loop! (line 25)


**Main**

> clear...

**HashmapInitilizer**

> -Class could be singleTone or have static members!

> -Class name doesnt follow cammel case pattern (HashMap!!)

> -"resources/SampleEnglishData" could be a private final static parameter (line 9)

> -proccessInfo() has logical issue! if the input be hello.toulab the word you are adding to your hashmap is hellotoulab instead of hello and toulab! (line 26)

**Comment by Arash1381-y** [comment]
@Siasor88 
thank u for your attention  
many of the problems you mention are fixed in new commits -> please take a look
the process of reaching a structure which can determine meaningful words or separating words are not the main goals of this phase as it mentioned by Our mentors but you are right and this separation is just for simple text with few grammatical problems

if the issues are fixed please don't forget to send  a feed back

**Pull request closed without merging**